### PR TITLE
Fix username retrieval

### LINKED ISSUE #42: Retrieve username from ssh_config
Credentials for `CredentialType::USERNAME` should be found from ssh_config.
I believe this can be done in the same way as for `CredentialType::SSH_KEY`.

### PR DESCRIPTION
**Pull request by gifnksm**
Fixes #42, #44

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -132,14 +121,15 @@ impl CredentialHandler {
         // handle this we bail out of this authentication session after setting
         // the flag `ssh_username_requested`, and then we handle this below.
         if allowed.contains(git2::CredentialType::USERNAME) {
-            // debug_assert!(username.is_none());
+            debug_assert!(username.is_none());
```
Comment (davidB) [inline src/lib.rs]: I don't want the lib to panic if `username` is defined. Can you comment the statement again or handle this case differently?

File: src/ssh_config.rs [hunk 2]
```diff
@@ -106,7 +128,7 @@ fn find_ssh_key_for_host_in_config(
                             ))
                         })?;
 
-                    if key.as_str().eq_ignore_ascii_case("IdentityFile") {
+                    if key.as_str().eq_ignore_ascii_case(name) {
                         let path = value.as_str().to_string();
 
                         // trace!("found IdentityFile option with value {:?}", path);
```
Comment (davidB) [inline src/ssh_config.rs]: As you modify the function (and this part) to not only retrieve a path of IdentityFile, can you also rename the variable `path` to reflect this change eg `found_value`).

Maybe in the future, an optimisation will be to not parse the file for every search.

**Review by davidB — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (gifnksm) [reply src/lib.rs]: Thank you for reviewing!

fixed by a2ee3078bcf63f806dfb639902727f4599bb5fd3

File: src/ssh_config.rs (hunk 2 above)
Comment (gifnksm) [reply src/ssh_config.rs]: fixed by c140c6794874453edc47b964f947ddd9c7b367d7

**Review by davidB — APPROVED** [review]

**Comment by davidB** [comment]
I release your fix as part of version 0.12.1 (for git2 0.17) and 0.13.0 (for git2 0.18).

Thank you.

**Pull request merged**